### DISCUSSION TITLE
Filament  V3 with Native PHP: issue with userMenuItems dropdown menu

### DISCUSSION THREAD

**Discussion by devhoussam1998** [filamentphp/filament#12388] [Help]
### Package

Panel builder

### Package Version

v3.2.70

### How can we help you?

there is a wrong dropdown menu direction

<img width="1107" alt="Screenshot 2024-04-18 at 15 28 14" src="https://github.com/filamentphp/filament/assets/121255432/dc24f2c8-05ec-4436-b805-bc4f61dbd8af">

**Comment by lcamero** [comment]
I recently came across an issue like this, probably because of the same
https://github.com/orgs/NativePHP/discussions/345

It has something to do with how the js-md5 package that filament uses identifies the NativePHP environment as NodeJS and fails to find Buffer

Not sure how to solve it

**Comment by bobicloudvision** [comment]
I have the same issue...

**Comment by simonhamp** [comment]
For anyone finding this wondering what the underlying problem is: thanks to @lcamero's digging and a very thorough investigation [here](https://qiita.com/nasteng/items/9984175ec505663c2321#%E3%82%B3%E3%83%A1%E3%83%B3%E3%83%88%E3%82%A2%E3%82%A6%E3%83%88%E3%81%97%E3%81%A6%E6%9C%AC%E5%BD%93%E3%81%AB%E5%A4%A7%E4%B8%88%E5%A4%AB%E3%81%AA%E3%81%AE%E3%81%8B), it seems this issue is to do with one of Filament's upstream dependencies ([`js-md5`](https://www.npmjs.com/package/js-md5)).

Filament team: I'd highly recommend either trying to find a way to MD5 in this context without that package. Alternatively, I've already [raised an issue with that package maintainer](https://github.com/emn178/js-md5/issues/47) on GitHub.

**Comment by simonhamp** [comment]
@devhoussam1998 @bobicloudvision @lcamero we've [had a suggestion](https://github.com/emn178/js-md5/issues/47#issuecomment-2339429516) from the `js-md5` maintainer:

Try setting `window.JS_MD5_NO_NODE_JS = true` somewhere in your javascript _before_ the Filament JS is loaded.

**Comment by lcamero** [comment]
@simonhamp That certainly seemed to do the trick, no errors in console and the dropdown works as expected.

Do you think this should be at least referenced in the NativePHP docs somewhere? It is not really an issue on that project so I don't know. 

For now, adding `window.JS_MD5_NO_NODE_JS = true` works 

<img width="724" alt="image" src="https://github.com/user-attachments/assets/1511bc39-7205-4b6f-ad0c-9e9fa97b27d2">

**Reply by simonhamp** [reply]
Yeh tricky one...

I think maybe a "known issues" segment in the `Debugging` section would be enough for now?

**Reply by simonhamp** [reply]
Somewhere in your layout _before_ the `@filamentScripts` are registered/loaded

**Reply by MACscr** [reply]
Are there any downsides to this workaround? Couldn't it be an environment check and added automatically when needed?

**Reply by simonhamp** [reply]
@MACscr downsides: I don't think so - you have this one boolean value set on your `window`. The only conflict I could foresee is if some other library you use somehow relies on this exact same variable's value being `false`.

Could it be added automatically: I guess it's possible. We could detect that you're using Filament by looking at your dependencies and then add this flag to the `window` object dynamically.

Happy to review a PR (in nativephp/laravel) for this if anyone wants to give it a go.

**Reply by muhghazaliakbar** [reply]
@simonhamp Could you tell me where the exact location where I should put the code? 🤔

**Reply by lcamero** [reply]
@muhghazaliakbar not sure if you're planning to add the PR, but in the meantime you could test this solution here on the discussion thread https://github.com/orgs/NativePHP/discussions/345#discussioncomment-11791395 if it is done from the PHP side

I have not tested myself but looks like it does the job

**Reply by muhghazaliakbar** [reply]
@lcamero Yes! it's working, thanks!

**Comment by raymondkingston** [comment]
For any future devs dealing with this problem...

I was having this same broken layout issue in my Filament 3 auth panel. I noticed if I refreshed the page, the layout would fix itself. Also determined the pattern of navigating from `/*` to `/auth/*` guaranteed the broken layout — broken until refresh or navigating _within_ `/auth/*`, that is. After hours of troubleshooting and learning it was a failure of proper Alpine/Livewire hydration and initialization, I found the culprit:

`wire:navigate`

If any of your non-Filament pages route to your Filament auth panel using `wire:navigate`, you'll get that broken layout on your first pageload in `/auth/*` due to partial page transitions, particularly when navigating between Livewire-inactive and Livewire-active pages.

Simply removing it did the trick. So changing it from
`<x-dropdown-link :href="route('edit-profile')" wire:navigate>`
to
`<x-dropdown-link :href="route('profile')">` 🤌

My app isn't a SPA, so `wire:navigate` isn't a requirement for me. If it is for you, good luck...

I hope this saves someone some time!
